python/ray/ml/config.py

```diff
@@ -105,6 +115,47 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         bundles = trainer_bundle + worker_bundles
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
+    @classmethod
+    def validate_config(
+        cls,
+        config: Union[ScalingConfig, "ScalingConfigDataClass"],
+        allowed_keys: Iterable[str],
+    ):
+        """
+        Validate config by raising an exception if any of allowed_keys
+        differs from the default value.
+
+        Args:
+            config: Config to check.
+            allowed_keys: ScalingConfigDataClass attribute keys that
+                can have a value different than the default one.
+        """
+        default_config = ScalingConfigDataClass()
+        if not isinstance(config, ScalingConfigDataClass):
+            config = ScalingConfigDataClass(**config)
+
+        allowed_keys = set(allowed_keys)
+
+        bad_allowed_keys = [key for key in allowed_keys if key not in config.__dict__]
```

Comment:
Nit: I was kinda confused by this at first. Could improve readability to name `config.__dict__` (e.g., `scaling_config_attributes = set(default_config.__dict__)` (not sure what the best variable name would be)

Reply:
Renamed to `invalid_keys`

Reply:
Actually we don't need this check at all. This is not a user-facing API and the invalid keys are static per Trainer. If we need this check it should be done somewhere else.